src/js/components/TaskDetail.js

```diff
@@ -151,56 +151,119 @@ class TaskSidePanelContents extends SidePanelContents {
     });
   }
 
-  getBasicInfo(task, node) {
-    // Hide when no task or when we are viewing debug tab
-    if (task == null || this.state.currentTab === 'debug') {
-      return null;
-    }
+  getBasicInfo(task) {
+    let taskIcon = (
+      <div
+        className="icon icon-large icon-image-container icon-app-container">
+        <img src={task.getImages()['icon-large']} />
```

Comment:
I'm not convinced that the `Task` struct should be responsible for holding an array of images. Is it the case that Structs generally are supposed to hold view-data? I personally would favour referencing the `ServiceImages` object from the component itself.